src/mame/video/tc0100scn.h

```diff
@@ -39,27 +41,16 @@ class tc0100scn_device : public device_t
 
 	/* Function to set separate color banks for the three tilemapped layers.
 	To change from the default (0,0,0) use after calling TC0100SCN_vh_start */
-	void set_colbanks(int bg0, int bg1, int tx);
-
-	/* Function to set bg tilemask < 0xffff */
-	void set_bg_tilemask(int mask);
-
-	/* Function to for Mjnquest to select gfx bank */
-	DECLARE_WRITE16_MEMBER(gfxbank_w);
+	void set_colbanks(u8 bg0, u8 bg1, u8 tx);
 
-	DECLARE_READ16_MEMBER(word_r);
-	DECLARE_WRITE16_MEMBER(word_w);
-	DECLARE_READ16_MEMBER(ctrl_word_r);
-	DECLARE_WRITE16_MEMBER(ctrl_word_w);
-
-	/* Functions for use with 68020 (Under Fire) */
-	DECLARE_READ32_MEMBER(long_r);
-	DECLARE_WRITE32_MEMBER(long_w);
-	DECLARE_READ32_MEMBER(ctrl_long_r);
-	DECLARE_WRITE32_MEMBER(ctrl_long_w);
+	u16 ram_r(offs_t offset, u16 mem_mask);
+	void ram_w(offs_t offset, u16 data, u16 mem_mask);
+	u16 ctrl_r(offs_t offset, u16 mem_mask);
+	void ctrl_w(offs_t offset, u16 data, u16 mem_mask);
```

Comment:
ram_r and ctrl_r are 16 bit read handlers, They working without data parameters in above.

Reply:
It will only work if you aren't actually using `mem_mask` for anything.  The memory system will call the handler with the offset and data, but now the mask.  The names of the parameters don't affect how the function is called, only the types.